c++/src/kj/async-inl.h

```diff
@@ -134,6 +134,11 @@ class TraceBuilder {
   void** limit;
 };
 
+struct alignas(void*) PromiseArena {
```

Comment:
I suppose it couldn't hurt to add `final` here, in case it maybe helps the compiler work out sizeof(PromiseArena) earlier?

Reply:
I don't think that makes a difference, since `PromiseArena` is not polymorphic (has no virtual methods), `final` doesn't enable any particular optimizations.

I did wonder for a bit if this has an effect on the compiler's ability to generate sized deletes (call the two-argument operator delete that takes the size as a parameter), but I couldn't actually get Clang to produce such calls in godbolt at all. Anyway, I don't think it should make a difference since it would be UB to delete a pointer to a superclass of an object if that superclass doesn't have a virtual destructor.

(Incidentally, I also tried directly calling two-argument operator delete, and even directly calling tcmalloc's underlying functions, but these both made things slower. I think the compiler must be able to optimize around operators new and delete when it actually understands them...)